data/latest/test-device-virtual.sh

```diff
@@ -62,7 +62,7 @@ while true; do
         print_error_logs
         echo "invalid JSON response from core-metadata"
         exit 1
-    elif [ "$(edgexfoundry.curl -s localhost:59881/api/v2/device/all | $JQ 'map(select(.name == "Random-Boolean-Device")) | length')" -lt 1 ]; then
+    elif [ "$(edgexfoundry.curl -s localhost:59881/api/v2/device/all | $JQ '[.devices[] | select(.name == "Random-Boolean-Device")] | length')" -lt 1 ]; then
```

Comment:
Line 60 prints the JSON object. Change to redirect to `/dev/null`.
Line 60 and 65 repeat the same query.

Actually, the first condition can be removed. The JSON object gets validated by jq anyway before performing the other operations and return the appropriate error.